scripts/kafka/install-kafka.sh

```diff
@@ -0,0 +1,274 @@
+#!/bin/sh
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+# Link for installation and configuration steps of kafka (single node and cluster) are available at https://github.com/Seagate/cortx-utils/wiki/Kafka-Server-Setup
+
+set -e
+
+#Variables
+KAFKA_INSTALL_PATH=/opt
+KAFKA_DOWNLOAD_URL="http://cortx-storage.colo.seagate.com/releases/cortx/third-party-deps/centos/centos-7.8.2003-1.0.0-3/commons/kafka/kafka_2.13-2.7.0.tgz"
+KAFKA_FOLDER_NAME="kafka"
+BGDELETE_TOPIC_NAME="bgdelete"
+consumer_count=0
+hosts=""
+bootstrapservers=""
+hostnumber=0
+ZOOKEEPER_FOLDER_NAME="zookeeper"
+
+# Function to install all pre-requisites
+install_prerequisite() {
+  echo "Installing Pre-requisites"
+  yum install java -y
+  yum install java-devel -y
+  echo "Pre-requisites installed successfully."
+}
+
+# function to download and setup kafka
+setup_kafka() {
+  echo "Installing and Setting up kafka."
+  cd $KAFKA_INSTALL_PATH
+  curl $KAFKA_DOWNLOAD_URL -o $KAFKA_FOLDER_NAME.tgz
+  if [ -d "$KAFKA_INSTALL_PATH/$KAFKA_FOLDER_NAME" ]; then
+    echo "kafka folder is already exist"
+  else
+    mkdir $KAFKA_FOLDER_NAME    
+  fi
+  tar -xzf $KAFKA_FOLDER_NAME.tgz -C $KAFKA_FOLDER_NAME --strip-components 1
+}
+
+#function to start services of kafka
+start_services() {
+  echo "Starting services..."
+  
+  cd $KAFKA_INSTALL_PATH/$KAFKA_FOLDER_NAME
+  
+  bin/zookeeper-server-stop.sh || true
+  #start zookeeper
+  nohup bin/zookeeper-server-start.sh config/zookeeper.properties &
```

Comment:
Should the correct order here be to check if both services are stopped? Then start zookeeper, then start kafka server?

Reply:
fixed